scripts/linux_share_build.sh

```diff
@@ -56,36 +58,65 @@ function build_dagmc(){
                 -DBUILD_EXE=OFF \
                 -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
                 -DCMAKE_BUILD_TYPE=Release \
+                $CMAKE_ADDITIONAL_FLAG \
                 -DCMAKE_INSTALL_PREFIX=${PLUGIN_ABS_PATH}/DAGMC
-    make -j$PROC
+                
+    
+    make -j`grep -c processor /proc/cpuinfo`
     make install
     cd ../..
-    rm -rf DAGMC/DAGMC DAGMC/bld
+    rm -rf DAGMC/DAGMC DAGCM/bld
 }
 
+
 function setup_Trelis_sdk() {
-    cd /Trelis-sdk 
-    dpkg -i Trelis-$1-Lin64.deb
+    if [ "$1" = "2020.2" ]; then
+        TRELIS_PATH="/opt/Coreform-Cubit-2020.2"
+        TRELIS_PKG="Coreform-Cubit-2020.2-Lin64.deb"
+        TRELIS_SDK_PKG="Coreform-Cubit-2020.2-Lin64-SDK.tar.gz"
+    elif [ "$1" = "17.1.0" ]; then
+        TRELIS_PATH="/opt/Trelis-17.1"
+        TRELIS_PKG="Trelis-17.1.0-Lin64.deb"
+        TRELIS_SDK_PKG="Trelis-SDK-17.1.0-Lin64.tar.gz"
+        CMAKE_ADDITIONAL_FLAG="-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=0"
+    else
+        echo "unknown Trelis/Cubit version, use: \"17.1\" or \"2020.2\""
+    fi
+    if [ "$1" = "2020.2" ]; then
+        apt-get -y install libxt6 libglew2.1 libglu1-mesa libopengl0 ocl-icd-libopencl1 libcxsparse3 libopenblas0 libegl1 libboost-thread1.71.0 liblapack3
+    fi
 
+    cd ${FOLDER_PKG} 
+    dpkg -i ${TRELIS_PKG}
     cd /opt
-    tar -xzvf /Trelis-sdk/Trelis-SDK-$1-Lin64.tar.gz
+    tar -xzvf ${FOLDER_PKG}/${TRELIS_SDK_PKG}
+    if [ "$1" = "2020.2" ]; then
+        cd ${TRELIS_PATH}/bin
+        cp -pv CubitExport.cmake CubitExport.cmake.orig
+        sed -i "s/\"\/\.\.\/app_logger\"/\"\"/" CubitExport.cmake
+        cp -pv CubitUtilConfig.cmake CubitUtilConfig.cmake.orig
+        sed -i "s/\/\.\.\/app_logger\;//" CubitUtilConfig.cmake
+    fi
```

Comment:
These regular expressions are one good example of why I think we should try to rely on the functions in these scripts in the GitHub actions and maintain only one version if possible.

Also, a comment as to what these lines are doing would be good.

Reply:
I agree and tried, but didn't found a way to satisfied me:
the problem is the GitHub Action job work on different terminal so local variable are not passed.

so one need to source the share_build at each steps
and when I update a variable value I need then to pass them to the system
I had the feeling that was getting too complicated.
(I though about removing the scripts, as the GitHub action were defined, but thought it is kinda nice to have them there.

Some thoughs
1.  maybe the complicated part could be extracted and shared between action and scripts.
2. maybe the docker script should not exist and the command line be mentioned in the README file

Reply:
Here is a though,
I adapt the `line_shared_build.sh` script/method to look at the state of a variable that in `ON` if we are in GitHub action. When GIT_ACTION_ENV is `ON`, the value of variables are sent to the GitHub Action env variable handler.
it adds a bit of complexity in the build_script but allow sharing the methods 

What do you think @pshriwise ?

Reply:
Yeah there are a few different ways we can handle those issues I think. Can we separate the environment into a different script that always gets sourced and ensures the environment is setup for any of the build steps regardless of docker, local, and gha? The `GHA_ACTION_ENV` sounds idea fine to me.

Reply:
Is this critical path for this PR?
or can it be a follow up?
Or are we worried it won't happen that way?

Reply:
I guess I think this is a good point to figure out how to rely on a common set of scripts that support all of these purposes. Otherwise, I think the code duplication in the GHA files is going to propagate faster than I'd like after looking ahead at #85 and #86.

If we handle this now, those PRs will be more focused on platform-specific issues than on new GHA files that look largely the same.

Reply:
Maybe those will still look different enough that it isn't worth it though? @bam241 would know better than I. It just feels off seeing all of the same CMake/make commands reappearing in all of these new `.yml` files.